lib/plug/adapters/cowboy2/handler.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Plug.Adapters.Cowboy2.Handler do
```

Comment:
Can we implement the early response check here instead of adding a custom handler? Maybe we cannot.